src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java

```diff
@@ -0,0 +1,96 @@
+
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.service.holder.ReferenceDataServiceHolder;
+
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.isYes;
+import static uk.gov.hmcts.reform.sscs.helper.mapping.HearingsCaseMapping.isInterpreterRequired;
+import static uk.gov.hmcts.reform.sscs.helper.mapping.HearingsDetailsMapping.getElementsDisputed;
+
+
+public final class HearingsDurationMapping {
+    public static final int DURATION_SESSIONS_MULTIPLIER = 165;
+    //public static final int DURATION_HOURS_MULTIPLIER = 60;
+    public static final int DURATION_DEFAULT = 30;
+    public static final int MIN_HEARING_DURATION = 30;
+
+    private HearingsDurationMapping() {
+
+    }
+
+    public static int getHearingDuration(SscsCaseData caseData, ReferenceDataServiceHolder referenceDataServiceHolder) {
+        OverrideFields overrideFields = OverridesMapping.getOverrideFields(caseData);
+
+        if (nonNull(overrideFields.getDuration()) && overrideFields.getDuration().intValue() >= MIN_HEARING_DURATION) {
+            return overrideFields.getDuration().intValue();
+        }
+
+        Integer duration = getHearingDurationAdjournment(caseData, referenceDataServiceHolder);
+        if (isNull(duration)) {
+            duration = getHearingDurationBenefitIssueCodes(caseData, referenceDataServiceHolder);
+        }
+
+        return nonNull(duration) ? duration : DURATION_DEFAULT;
+    }
+
+    public static Integer getHearingDurationAdjournment(SscsCaseData caseData, ReferenceDataServiceHolder referenceDataServiceHolder) {
+        final String nonStandardTimeSlot = "nonStandardTimeSlot";
+        if (nonStandardTimeSlot.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationType())
+            && isNotBlank(caseData.getAdjournCaseNextHearingListingDuration())) {
+            int duration =  Integer.parseInt(caseData.getAdjournCaseNextHearingListingDuration());
+            boolean isSessions = false;
+            final String sessions = "sessions";
+            if (sessions.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())) {
+                duration = duration * DURATION_SESSIONS_MULTIPLIER;
+                isSessions = true;
+            }
+            final String minutes = "minutes";
+            if ((minutes.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits()) || isSessions)
+                && duration >= MIN_HEARING_DURATION) {
+                return duration;
+            }
+        }
+
+        if ("standardTimeSlot".equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationType())) {
+            return getHearingDurationBenefitIssueCodes(caseData, referenceDataServiceHolder);
+        }
```

Comment:
You don't need these lines, a standard time slot if dealt with by the parent method

Reply:
You mean lines 60 - 62, this method is public static, will it always be called by getHearingDuration()?

Reply:
Done